uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs

```diff
@@ -162,37 +369,98 @@ createSkipOverride ::
+  Maybe (Either (ClobberSpecError m) (PolymorphicLLVMOverride arch (personality sym) sym))
+createSkipOverride modCtx sym usedRef annotationRef clobbers postcondition funcSymb =
   llvmDeclToFunHandleRepr' decl $
-    \args ret ->
+    \argTys retTy ->
       Just $
-        makePolymorphicLLVMOverride $
-          basic_llvm_override $
-            LLVMOverride
-              { llvmOverride_declare = decl,
-                llvmOverride_args = args,
-                llvmOverride_ret = ret,
-                llvmOverride_def =
-                  \mvar _sym _args ->
-                    do
-                      liftIO $
-                        modifyIORef usedRef (Set.insert (SkipOverrideName name))
-                      Override.modifyGlobal mvar $ \mem ->
-                        liftIO
-                          ( returnValue
-                              mem
-                              ret
-                              (modCtx ^. funcTypes . funcSymbol funcSym)
-                          )
-              }
+        case makeGetters modCtx funcSymb argTys clobbers of
+          Left err -> Left err
+          Right clobbers' ->
+            Right $
+              makePolymorphicLLVMOverride $
+                basic_llvm_override $
+                  LLVMOverride
+                    { llvmOverride_declare = decl,
+                      llvmOverride_args = argTys,
+                      llvmOverride_ret = retTy,
+                      llvmOverride_def =
+                        \mvar _sym args ->
+                          do
+                            liftIO $
+                              modifyIORef usedRef (Set.insert (SkipOverrideName name))
+                            Override.modifyGlobal mvar $
+                              \mem ->
+                                liftIO $
+                                  do mem' <- clobberAll mem args clobbers'
+                                     returnValue
+                                       mem'
+                                       retTy
+                                       (modCtx ^. funcTypes . funcSymbol funcSymb)
+                    }
   where
+    decl = modCtx ^. moduleDecls . funcSymbol funcSymb
     name = declName decl
     symbolName = L.decName decl
 
+    clobberOne ::
+      MemImpl sym ->
+      Ctx.Assignment (Crucible.RegEntry sym) args ->
+      ClobberSpec m args' inTy atTy ->
+      ClobberGetter arch args inTy ->
+      IO (MemImpl sym)
+    clobberOne mem args spec getter =
+      let mts = modCtx ^. moduleTypes
+      in
+        runSetup
+          modCtx
+          mem
+          ( generate
+              sym
+              modCtx
+              (clobberAtType mts spec)
+              (SelectClobbered
+                 funcSymb
+                 (deepenPtr mts (clobberSelectorCursor (clobberSelector spec))))
+              (clobberShape spec)
+          )
+          >>=
+            \case
+              (result, value) ->
```

Comment:
Isn't this just `\(result, value) ->`?

Reply:
Also does it need to be one line? There is already a nice readable `do` block below. Why not just use monadic syntax with the first bit, too?

Reply:
This is cruft from when this used to return an `Either`. Fixed, good catch!